packages/core/src/index.ts

```diff
@@ -74,6 +75,22 @@ class ImageService extends Service {
 
     return undefined
   }
+
+  async imgUrlToAssetUrl(image) {
+    return 'http://' + await this.ctx.assets.upload(image.url, process.uptime().toString())
+  }
+
+  async imgUrlToBase64(image) {
+    const buffer = await this.ctx.http.get(image.url, { responseType: 'arraybuffer' }).catch((err) => {
+      if (Quester.isAxiosError(err)) {
+        logger.warn(`request failed when switch a iamge to base64 format with code ${err.status} ${JSON.stringify(err.response?.data)}`)
+      } else {
+        logger.error(`unknown error when switch a iamge to base64 format: ${err.message}`)
+      }
+      return ''
+    })
+    return 'data:image/*;base64,' + Buffer.from(buffer, 'binary').toString('base64')
```

Comment:
This is not correct, you should always return a valid mime tag for the current image, which you could take from the response header `Content-Type` for example.

Reply:
But when using an `asset` service, could it return a correct response header?

Reply:
Most importantly, the `axios.get` we use cannot view the response header.

Reply:
Change to 
```js
const res = ctx.http.axios(url, { method: 'GET' })
res.header['Context-type'] // -> image/xxx
```

Reply:
Outdated.